test/e2e/test/agent/builder.go

```diff
@@ -195,13 +197,31 @@ func (b Builder) WithRestrictedSecurityContext() Builder {
 }
 
 func (b Builder) WithContainerSecurityContext(securityContext corev1.SecurityContext) Builder {
-	for i := range b.PodTemplate.Spec.Containers {
-		b.PodTemplate.Spec.Containers[i].SecurityContext = &securityContext
+	containerIdx := getContainerIndex(agent.ContainerName, b.PodTemplate.Spec.Containers)
+	if containerIdx < 0 {
+		b.PodTemplate.Spec.Containers = append(
+			b.PodTemplate.Spec.Containers,
+			corev1.Container{
+				Name:            agent.ContainerName,
+				SecurityContext: &securityContext,
+			},
+		)
+		return b
 	}
 
+	b.PodTemplate.Spec.Containers[containerIdx].SecurityContext = &securityContext
```

Comment:
Shouldn't we also fix [`WithContainerSecurityContext`](https://github.com/elastic/cloud-on-k8s/blob/4233f7b8e6bc8dc90712cb65c12dc0c89bfcc8f1/test/e2e/test/beat/builder.go#L208) in the `Beat` builder?

Reply:
> Shouldn't we also fix [WithContainerSecurityContext](https://github.com/elastic/cloud-on-k8s/blob/4233f7b8e6bc8dc90712cb65c12dc0c89bfcc8f1/test/e2e/test/beat/builder.go#L208) in the Beat builder?

It does not seem to be used, I removed it.